core/src/main/java/c/org/rajawali3d/surface/gles/BaseGlesSurfaceRenderer.java

```diff
@@ -1,30 +1,35 @@
 package c.org.rajawali3d.surface.gles;
 
+import c.org.rajawali3d.control.CoreControl;
+import c.org.rajawali3d.control.RenderContextType;
+import c.org.rajawali3d.control.RenderControlClient;
+import c.org.rajawali3d.control.RenderSurfaceView;
+import c.org.rajawali3d.control.gles.GlesRenderControl;
+import c.org.rajawali3d.gl.Capabilities;
+import c.org.rajawali3d.surface.SurfaceRenderer;
+
 import android.content.Context;
 import android.opengl.GLES20;
-import c.org.rajawali3d.core.ARenderControl;
-import c.org.rajawali3d.core.RenderContextType;
-import c.org.rajawali3d.core.RenderControlClient;
-import c.org.rajawali3d.core.RenderSurfaceView;
-import c.org.rajawali3d.gl.Capabilities;
+
 import org.rajawali3d.util.RajLog;
 
 import java.util.Locale;
 
 /**
- * Shared implementation for GL ES extensions of {@link ARenderControl}
+ * Shared implementation for GLES {@link SurfaceRenderer} extensions of {@link CoreControl}
  *
  * @author Randy Picolet
  */
-abstract class GLESRenderer extends ARenderControl {
 
-    GLESRenderer(Context context, RenderSurfaceView renderSurfaceView,
-                 RenderControlClient renderControlClient, double initialFrameRate) {
+abstract class BaseGlesSurfaceRenderer extends CoreControl implements GlesRenderControl {
+
+    BaseGlesSurfaceRenderer(Context context, RenderSurfaceView renderSurfaceView,
+                            RenderControlClient renderControlClient, double initialFrameRate) {
         super(context, renderSurfaceView, renderControlClient, initialFrameRate);
     }
 
     /**
-     * Callback to notify that the EGL context has been acquired and a valid GL thread with EGL surface now exists.
+     *
      */
     protected void onRenderContextAcquired() {
```

Comment:
Don't omit `@Override` annotations.

Reply:
Never on purpose, always flag em for me.

Reply:
Although in this case its just the shared implementation, not overriding anything...  Will document

Reply:
What do you mean by shared implementation?

Reply:
This method is used by both GL surface view renderers to get the context version number before forwarding up to `CoreControl.onRenderContextAcquired()`. Didn't want to duplicate the code in both places is all.

Reply:
The problem is it implements an interface which defines a function of that name. Intentional or not, you overrode it.

Reply:
The interface method is public and has context type/version parameters, this one is protected and has none. Different signatures. This one wraps/calls the interface method on behalf of both surface views, since they both figure out the version the same way; it does not override or extend the interface method.

Reply:
Then I would argue one of them needs a different name.

Reply:
That's legit, renamed it to `notifyRenderContextAcquired`.